File: packages/client-java/src/main/java/com/durablestreams/exception/StreamClosedException.java
package com.durablestreams.exception;

/**
 * Exception thrown when attempting to append to a closed stream.
 */
public class StreamClosedException extends DurableStreamException {

    public StreamClosedException(String url) {
        super("Stream is closed: " + url, 409);
    }

    public StreamClosedException(String message, int statusCode) {
        super(message, statusCode);
    }
}

File: packages/durable-streams/packages/client-php/src/Exception/StreamClosedException.php
<?php

declare(strict_types=1);

namespace DurableStreams\Exception;

/**
 * Exception thrown when attempting to append to a closed stream.
 */
class StreamClosedException extends DurableStreamException
{
    public function __construct(string $url = null, ?\Throwable $previous = null)
    {
        $message = $url ? "Stream is closed: {$url}" : "Stream is closed";
        parent::__construct($message, 'STREAM_CLOSED', 409, [], $previous);
    }
}
